content/post/bartlombardi/20/git-cherry-pick-for-a-strategic-harvest/index.en.md

```diff
@@ -0,0 +1,52 @@
+---
+title: "Git Cherry-Pick for a strategic harvest"
+date: 2020-07-13T03:00:00+02:00
+authors: ['bartlombardi']
+categories:
+- devops articles
+tags:
+- azure devops
+- git
+- cherry-pick
+summary: "Fixing a bug on multiple branches is a complex task. In this article I will illustrate the Git cherry-pick command."
+publishDate: 2020-08-28T03:00:00+02:00
+---
+
+# Introduction
+
+The IT companies that develop and sell software product are always focused on delivering new features to propose new versions of them to the final client, still considering the maintenance of the previous release.
+Generally, we are talking about projects of millions of lines of code that requires multiple developing teams located around the world with strong coordination and collaboration among them, in which it is necessary to handle multiple versions of the same product with a significant number of release branches. Fixing a bug on multiple branches is a complex task because the developer has to report the same corrections in the various release branches. This action is possible in Git and it is called "cherry-picking".
+
+# Git cherry-pick
+
+One of the most powerful commands available in Git is Cherry-pick. It takes as inputs one or more commits and applies them to a different branch, by creating a new commit. 
+
+Cherry-pick is commonly used in many Git workflows such as the Azure DevOps team’s, described in the following [article](https://devblogs.microsoft.com/devops/improving-azure-devops-cherry-picking/), see the image reported below for a representation of the workflow.
+![Cherry-Pick: way of working](cherry-pick-workflow.jpg)
+
+Anytime a developer works with many version of the same product it is essential to assure that all the bug reported have been fixed in every version. The cherry-picking action is performed into the main branch also to avoid deploying a new release with the same bug.
```

Comment:
Every time a developer works with many version of the same product it is essential to ensure all the reported bugs have been fixed in as many version as possible. Cherry-picking happens on the main branch to avoid deploying a new release with the same bug.